Bug fixes to the workflow xml files, add additional workflow steps for min/maxt t obs and bufr sounding profiles.

EdwardColon-NOAA opened the pull request:
…r min/maxt t obs and bufr sounding profiles.

EdwardColon-NOAA committed:
Bug fixes to the workflow xml files, add additional workflow steps for min/maxt t obs and bufr sounding profiles.

GangZhao-NOAA commented:
@EdwardColon-NOAA 
Hi Ed,

I am reviewing the changed files now. Just to confirm with you that, you already tested this modified workflow in real-time run for a few cycles on wcoss2m, right?  Because we do not have ctest or regression test for this system. 

Thanks,
-Gang

EdwardColon-NOAA commented:
Yes. I did test the changes before issuing the PR.

-Edward

-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566


On Fri, Feb 28, 2025 at 8:50 AM Gang Zhao ***@***.***> wrote:

> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>
> Hi Ed,
>
> I am reviewing the changed files now. Just to confirm with you that, you
> already tested this modified workflow in real-time run for a few cycles on
> wcoss2m, right? Because we do not have ctest or regression test for this
> system.
>
> Thanks,
> -Gang
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/30#issuecomment-2690697890>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQD324EKGOHU36UZ4Z32SBSRZAVCNFSM6AAAAABYBOVLBWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJQGY4TOOBZGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: GangZhao-NOAA]*GangZhao-NOAA* left a comment
> (NOAA-EMC/NOAA_3drtma#30)
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/30#issuecomment-2690697890>
>
> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>
> Hi Ed,
>
> I am reviewing the changed files now. Just to confirm with you that, you
> already tested this modified workflow in real-time run for a few cycles on
> wcoss2m, right? Because we do not have ctest or regression test for this
> system.
>
> Thanks,
> -Gang
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/30#issuecomment-2690697890>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQD324EKGOHU36UZ4Z32SBSRZAVCNFSM6AAAAABYBOVLBWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJQGY4TOOBZGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

EdwardColon-NOAA committed:
Update build_rtma_all.sh

Updated the cpreq command to cp.

MatthewMorris-NOAA reviewed (approved):
Just a note that these lines will eventually need to be removed for NCO compliance. The release notes will need to direct the SPA where to get these files.

GangZhao-NOAA commented on scripts/akrtma3d/exakrtma3d_gsianl.ksh:
```diff
@@ -47,7 +47,7 @@ time_str2=`${DATE} "+%Y-%m-%d_%H_00_00" -d "${START_TIME}"`
 cd ${DATA}
 ${ECHO} "enter working directory:${DATA}"
-nc_diag_cat=/lfs/h2/emc/da/noscrub/edward.colon/save/RTMA/bin/ncdiag_cat_serial.x
+nc_diag_cat=${EXECrtma3d}/ncdiag_cat_serial.x
```
@EdwardColon-NOAA 
Hi Ed,
${EXECrtma3d} is the ${HOMErtma3d}/exec directory, where the compiled exe will be stored. So the executable ncdiag_cat_serial.x is also built when building the whole 3DRTMA package? Have you include the source code of ncdiag in 3DRTMA package? But I did not read the lines to build ncdiag in build_rtam_all.sh.

Annette (@AnnetteGibbs-NOAA ) suggested to load ncdiag module, then the executable of ncdiag could be used directly.

Or you may add a line in "build_rtma_all.sh" to copy your pre-compiled ncdaig exe to exec/, something like 
`cp -p /lfs/h2/emc/da/noscrub/edward.colon/NOAA_3drtma_new/exec/ncdiag_cat_serial.x   ${BASE}../exec `

Thanks,
-Gang

GangZhao-NOAA commented on scripts/rtma3d/exrtma3d_gsianl.ksh:
```diff
@@ -47,7 +47,7 @@ time_str2=`${DATE} "+%Y-%m-%d_%H_00_00" -d "${START_TIME}"`
 cd ${DATA}
 ${ECHO} "enter working directory:${DATA}"
-nc_diag_cat=/lfs/h2/emc/da/noscrub/edward.colon/save/RTMA/bin/ncdiag_cat_serial.x
+nc_diag_cat=${EXECrtma3d}/ncdiag_cat_serial.x
```
@EdwardColon-NOAA 
Hi Ed,

Similar as my questions and comments above in "scripts/akrtma3d/exakrtma3d_gsianl.ksh". We need to build the exe ncdiag_cat_serial.x when building  the 3DRTMA pacakge, and we also need to include the source code of ncdiag in the pacakge.   Or just as Annette mentioned, loading mcdiag module might be alternative.

or as I mentioned above, add a line in building script
`cp -p /lfs/h2/emc/da/noscrub/edward.colon/NOAA_3drtma_new/exec/ncdiag_cat_serial.x   ${BASE}../exec `

Thanks,
-Gang

AnnetteGibbs-NOAA replied:
@GangZhao-NOAA I plan to load the ncdiag module.  I added it to the gsi script for my NCO-compliancy work, but did not have a chance to test it yet.  I should point out that RRFS is loading the ncdiag module instead of including the ncdiag code.

GangZhao-NOAA commented on scripts/akrtma3d/exakrtma3d_gsianl.ksh:
```diff
@@ -122,7 +122,7 @@ if [ "${envir}" = "lsf" ] || [ "${envir}" = "pbspro" ] && [ ${HRRRDAS_BEC} -eq 0
   ##typeset -Z2 nhr_assimilation
 
-  /usr/bin/python ${UTILrtma3d_dev}/getbest_EnKF_FV3GDAS.py -v $YYYYMMDDHH --exact=no --minsize=${nens} -d ${COMINGDAS}/enkfgdas -m no -o filelist${nhr_assimilation} --o3fname=gfs_sigf${nhr_assimilation} --gfs_netcdf=yes   
+  python ${UTILrtma3d_dev}/getbest_EnKF_FV3GDAS.py -v $YYYYMMDDHH --exact=no --minsize=${nens} -d ${COMINGDAS}/enkfgdas -m no -o filelist${nhr_assimilation} --o3fname=gfs_sigf${nhr_assimilation} --gfs_netcdf=yes   
```
@EdwardColon-NOAA 
Hi Ed,

The absolute path to python (/usr/bin/) was removed here, then OS needs to search for "python". Then python module should be loaded first so that OS could find python. Otherwise, if no python module is loaded or no specific PYTHONPATH is defined in advance, then OS would search in the default PATH.

I made a few tests on WCOSS2 (also @AnnetteGibbs-NOAA), if not loading any python model, then even if removing absolute path "/usr/path/", OS still finds "/usr/bin/python" almost every time (and it is linked to python2.7). Just in very few cases (1 out of 20 or even less) OS could not find python under "/usr/bin/" (as we discussed in our email about the weird behavior on wcsso2), I do not know why, maybe it is due to the file system glitch, or python is missing on a very few nodes.

So this is just a comment, you can keep use python is this way (removing the absolute path "/usr/bin/", OS should be able to find the default python under /usr/bin in almost all cases.

And for consistency, you might need to do the same thing for "/usr/bin/python" in "scripts/rtma3d/exrtma3d_gsianl.ksh" . (I cannot make a comment in that script for that line 144 "/usr/bin/python", because you did not make change to that line.)

Thank you!

-Gang

AnnetteGibbs-NOAA replied:
@GangZhao-NOAA I wonder if python is missing on a few nodes.  I will check all my log files to see if the missing python has occurred again and if it has, determine if it is the same node(s).  Also, I plan to remove the call to the python script as part of my NCO-compliancy work.

GangZhao-NOAA reviewed (changes requested):
@EdwardColon-NOAA,
Hi Ed,
For max/minT and bufrsnd part, I am not familiar with them. 
I made two comments for same issue in "scripts/akrtma3d/exakrtma3d_gsianl.ksh" and "scripts/rtma3d/exrtma3d_gsianl.ksh", for the same line
nc_diag_cat=${EXECrtma3d}/ncdiag_cat_serial.x

If ncdiag is not built in building script, and no ncdiag module would be loaded, could you add a line in the build_rtma_all.sh
`cp -p /lfs/h2/emc/da/noscrub/edward.colon/NOAA_3drtma_new/exec/ncdiag_cat_serial.x   ${BASE}../exec `

(when Annette added the line in running modulefile to load ncdiag module, that line could be removed in the build_rtma_all.sh)

As the comments for changing "/usr/bin/python" to "python" in "scripts/akrtma3d/exakrtma3d_gsianl.ksh", it is fine with me. But for consistency, you might need to do the same in "scripts/rtma3d/exrtma3d_gsianl.ksh"

Thank you!

-Gang

EdwardColon-NOAA commented:
Hi Gang,

I am leaving for the airport but will address these issues when I return
Monday.

-Edward

On Fri, Feb 28, 2025 at 1:55 PM Gang Zhao ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>,
> Hi Ed,
> For max/minT and bufrsnd part, I am not familiar with them.
> I made two comments for same issue in
> "scripts/akrtma3d/exakrtma3d_gsianl.ksh" and
> "scripts/rtma3d/exrtma3d_gsianl.ksh", for the same line
> nc_diag_cat=${EXECrtma3d}/ncdiag_cat_serial.x
>
> If ncdiag is not built in building script, and no ncdiag module would be
> loaded, could you add a line in the build_rtma_all.sh
> cp -p
> /lfs/h2/emc/da/noscrub/edward.colon/NOAA_3drtma_new/exec/ncdiag_cat_serial.x
> ${BASE}../exec
>
> (when Annette added the line in running modulefile to load ncdiag module,
> that line could be removed in the build_rtma_all.sh)
>
> As the comments for changing "/usr/bin/python" to "python" in
> "scripts/akrtma3d/exakrtma3d_gsianl.ksh", it is fine with me. But for
> consistency, you might need to do the same in
> "scripts/rtma3d/exrtma3d_gsianl.ksh"
>
> Thank you!
>
> -Gang
> ------------------------------
>
> In scripts/akrtma3d/exakrtma3d_gsianl.ksh
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/30#discussion_r1975823339>:
>
> > @@ -47,7 +47,7 @@ time_str2=`${DATE} "+%Y-%m-%d_%H_00_00" -d "${START_TIME}"`
>  #----- enter working directory -------
>  cd ${DATA}
>  ${ECHO} "enter working directory:${DATA}"
> -nc_diag_cat=/lfs/h2/emc/da/noscrub/edward.colon/save/RTMA/bin/ncdiag_cat_serial.x
> +nc_diag_cat=${EXECrtma3d}/ncdiag_cat_serial.x
>
> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>
> Hi Ed,
> ${EXECrtma3d} is the ${HOMErtma3d}/exec directory, where the compiled exe
> will be stored. So the executable ncdiag_cat_serial.x is also built when
> building the whole 3DRTMA package? Have you include the source code of
> ncdiag in 3DRTMA package? But I did not read the lines to build ncdiag in
> build_rtam_all.sh.
>
> Annette ***@***.*** <https://github.com/AnnetteGibbs-NOAA> )
> suggested to load ncdiag module, then the executable of ncdiag could be
> used directly.
>
> Thanks,
> -Gang
> ------------------------------
>
> In scripts/rtma3d/exrtma3d_gsianl.ksh
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/30#discussion_r1975830121>:
>
> > @@ -47,7 +47,7 @@ time_str2=`${DATE} "+%Y-%m-%d_%H_00_00" -d "${START_TIME}"`
>  #----- enter working directory -------
>  cd ${DATA}
>  ${ECHO} "enter working directory:${DATA}"
> -nc_diag_cat=/lfs/h2/emc/da/noscrub/edward.colon/save/RTMA/bin/ncdiag_cat_serial.x
> +nc_diag_cat=${EXECrtma3d}/ncdiag_cat_serial.x
>
> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>
> Hi Ed,
>
> Similar as my questions and comments above in
> "scripts/akrtma3d/exakrtma3d_gsianl.ksh". We need to build the exe
> ncdiag_cat_serial.x when building the 3DRTMA pacakge, and we also need to
> include the source code of ncdiag in the pacakge. Or just as Annette
> mentioned, loading mcdiag module might be alternative.
>
> Thanks,
> -Gang
> ------------------------------
>
> In scripts/akrtma3d/exakrtma3d_gsianl.ksh
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/30#discussion_r1975859456>:
>
> > @@ -122,7 +122,7 @@ if [ "${envir}" = "lsf" ] || [ "${envir}" = "pbspro" ] && [ ${HRRRDAS_BEC} -eq 0
>    export nhr_assimilation=03
>    ##typeset -Z2 nhr_assimilation
>
> -  /usr/bin/python ${UTILrtma3d_dev}/getbest_EnKF_FV3GDAS.py -v $YYYYMMDDHH --exact=no --minsize=${nens} -d ${COMINGDAS}/enkfgdas -m no -o filelist${nhr_assimilation} --o3fname=gfs_sigf${nhr_assimilation} --gfs_netcdf=yes
> +  python ${UTILrtma3d_dev}/getbest_EnKF_FV3GDAS.py -v $YYYYMMDDHH --exact=no --minsize=${nens} -d ${COMINGDAS}/enkfgdas -m no -o filelist${nhr_assimilation} --o3fname=gfs_sigf${nhr_assimilation} --gfs_netcdf=yes
>
> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>
> Hi Ed,
>
> The absolute path to python (/usr/bin/) was removed here, then OS needs to
> search for "python". Then python module should be loaded first so that OS
> could find python. Otherwise, if no python module is loaded or no specific
> PYTHONPATH is defined in advance, then OS would search in the default PATH.
>
> I made a few tests on WCOSS2 (also @AnnetteGibbs-NOAA
> <https://github.com/AnnetteGibbs-NOAA>), if not loading any python model,
> then even if removing absolute path "/usr/path/", OS still finds
> "/usr/bin/python" almost every time (and it is linked to python2.7). Just
> in very few cases (1 out of 20 or even less) OS could not find python under
> "/usr/bin/" (as we discussed in our email about the weird behavior on
> wcsso2), I do not know why, maybe it is due to the file system glitch, or
> python is missing on a very few nodes.
>
> So this is just a comment, you can keep use python is this way (removing
> the absolute path "/usr/bin/", OS should be able to find the default python
> under /usr/bin in almost all cases.
>
> And for consistency, you might need to do the same thing for
> "/usr/bin/python" in "scripts/rtma3d/exrtma3d_gsianl.ksh" . (I cannot make
> a comment in that script for that line 144 "/usr/bin/python", because you
> did not make change to that line.)
>
> Thank you!
>
> -Gang
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/30#pullrequestreview-2651461901>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQCB5PAXFE6UFN54RFD2SCWJ5AVCNFSM6AAAAABYBOVLBWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMNJRGQ3DCOJQGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

GangZhao-NOAA commented:
@EdwardColon-NOAA 
Hi Ed,
No problem. Have a good trip and a nice weekend!
Best regards,
-Gang

EdwardColon-NOAA committed:
Update build_rtma_all.sh

added wrfbufr fix files and an adhoc copy command of the ncdiag_cat_serial.exe executable to the user's exe directory.

EdwardColon-NOAA committed:
Update exrtma3d_gsianl.ksh

removed the the /usr/bin path for the python call as it does not appear to be a stable command.

AnnetteGibbs-NOAA commented on scripts/rtma3d/exrtma3d_gsianl.ksh:
```diff
@@ -177,7 +196,7 @@ if [[ ${hrrrmem} -gt 30 ]] && [[ ${HRRRDAS_BEC} -eq 1  ]]; then #if HRRRDAS BEC
   cpreq filelist.hrrrdas filelist03
   ${CP} ${PARMgsi}/hybens_info_hrrrdas hybens_info
-  beta1_inv=$(( 1 - $EnsWgt  ))
+  beta1_inv=0.9
```
@EdwardColon-NOAA Why did you change beta1_inv to 0.9?It was set to 0.1  (1 - 0.9 = 0.1)

EdwardColon-NOAA replied:
Hi Annette. Thank you for the correction. I'll address this now.

EdwardColon-NOAA committed:
Update exakrtma3d_gsianl.ksh

Resolved discrepancies in the value of beta_inv.

EdwardColon-NOAA committed:
Update exrtma3d_gsianl.ksh

Resolved discrepancies in the value of beta_inv.
